restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java

```diff
@@ -1361,8 +1362,16 @@ public void onConnectivityEvent(String jobId, RCDeviceListener.RCConnectivitySta
    {
       RCLogger.i(TAG, "onConnectivityEvent(): id: " + jobId + ", connectivityStatus: " + connectivityStatus);
       cachedConnectivityStatus = connectivityStatus;
-      if (state == DeviceState.OFFLINE && connectivityStatus != RCDeviceListener.RCConnectivityStatus.RCConnectivityStatusNone) {
-         state = DeviceState.READY;
+
+      storageManagerPreferences = new StorageManagerPreferences(this);
+      boolean pushEnabled = storageManagerPreferences.getBoolean(ParameterKeys.PUSH_NOTIFICATIONS_ENABLE_PUSH_FOR_ACCOUNT, false);
+      String binding = storageManagerPreferences.getString(FcmConfigurationHandler.FCM_BINDING, null);
```

Comment:
Are we sure when an FCM binding is found it means we are registered properly? Wanna make sure there's no chance a value has stayed there, but still last push registration failed

Reply:
I have changed inside FCMHandler if error happens, clear all cached data related for push.
Check line: FcmConfigurationHandler line 382.